go/porcelain/deploy.go

```diff
@@ -633,43 +635,51 @@ func bundle(functionDir string, observer DeployObserver) (*deployFiles, error) {
 		}
 	}
 
-	return functions, nil
+	return functions, nil, nil
 }
 
-func bundleFromManifest(manifestFile *os.File, observer DeployObserver) (*deployFiles, error) {
+func bundleFromManifest(manifestFile *os.File, observer DeployObserver) (*deployFiles, []models.FunctionSchedule, error) {
 	manifestBytes, err := ioutil.ReadAll(manifestFile)
 
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	var manifest functionsManifest
 
 	err = json.Unmarshal(manifestBytes, &manifest)
 
 	if err != nil {
-		return nil, fmt.Errorf("malformed functions manifest file: %w", err)
+		return nil, nil, fmt.Errorf("malformed functions manifest file: %w", err)
 	}
 
+	schedules := make([]models.FunctionSchedule, 0, len(manifest.Functions))
 	functions := newDeployFiles()
 
 	for _, function := range manifest.Functions {
 		fileInfo, err := os.Stat(function.Path)
 
 		if err != nil {
-			return nil, fmt.Errorf("manifest file specifies a function path that cannot be found: %s", function.Path)
+			return nil, nil, fmt.Errorf("manifest file specifies a function path that cannot be found: %s", function.Path)
 		}
 
 		file, err := newFunctionFile(function.Path, fileInfo, function.Runtime, observer)
 
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+
+		if function.Schedule != "" {
```

Comment:
today I learned that Golang's JSON unmarshaling uses the empty string as placeholder if a field doesn't exist

Reply:
That's actually not specific to JSON unmarshalling. When you initialise a variable with a struct type, the properties will use the corresponding [zero values](https://go.dev/tour/basics/12) for their type, which for strings is `""`.

When doing the unmarshalling, any properties that don't exist in the input data will be ignored, so the result will be whatever value was already there. If we wanted to initialise `function.Schedule` with a different value, that value would be the one left in the struct after doing the unmarshalling.